homeassistant/components/waze_travel_time/__init__.py

```diff
@@ -147,38 +153,92 @@ async def async_get_travel_times(
             alternatives=3,
         )
+        _LOGGER.debug("Got routes: %s", routes)
 
-        if incl_filter not in {None, ""}:
-            routes = [
-                r
-                for r in routes
-                if any(
-                    incl_filter.lower() == street_name.lower()  # type: ignore[union-attr]
-                    for street_name in r.street_names
-                )
-            ]
-
-        if excl_filter not in {None, ""}:
-            routes = [
-                r
-                for r in routes
-                if not any(
-                    excl_filter.lower() == street_name.lower()  # type: ignore[union-attr]
-                    for street_name in r.street_names
-                )
-            ]
+        incl_routes: list[CalcRoutesResponse] = []
 
-        if len(routes) < 1:
+        def should_include_route(route: CalcRoutesResponse) -> bool:
+            if len(incl_filters) < 1:
+                return True
+            should_include = any(
+                street_name in incl_filters or "" in incl_filters
+                for street_name in route.street_names
+            )
+            if not should_include:
+                _LOGGER.debug(
+                    "Excluding route [%s], because no inclusive filter matched any streetname",
+                    route.name,
+                )
+                return False
+            return True
+
+        incl_routes = [route for route in routes if should_include_route(route)]
+
+        filtered_routes: list[CalcRoutesResponse] = []
+
+        def should_exclude_route(route: CalcRoutesResponse) -> bool:
+            for street_name in route.street_names:
+                for excl_filter in excl_filters:
+                    if excl_filter == street_name:
+                        _LOGGER.debug(
+                            "Excluding route, because exclusive filter [%s] matched streetname: %s",
+                            excl_filter,
+                            route.name,
+                        )
+                        return True
+            return False
+
+        filtered_routes = [
+            route for route in incl_routes if not should_exclude_route(route)
+        ]
+
+        if len(filtered_routes) < 1:
             _LOGGER.warning("No routes found")
             return None
     except WRCError as exp:
         _LOGGER.warning("Error on retrieving data: %s", exp)
         return None
 
     else:
-        return routes
+        return filtered_routes
 
 
 async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(config_entry, PLATFORMS)
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate an old config entry."""
+
+    for migration_func in (_migrate_incl_excl_filters,):
+        if not migration_func(hass, config_entry):
+            return False
```

Comment:
Should we use the same pattern as we do at other places? this makes it less intuitive that its a version based migration

Reply:
I copied this from an existing approach (forgot which integration). Can you provide a link to where the pattern you have in mind is applied?

Reply:
https://github.com/home-assistant/core/blob/a89dcbc78b210035aa83919f03a0677aa6995c35/homeassistant/components/forecast_solar/__init__.py#L22-L37

When you view this one you can see like "oh this happens at version 1, and this at version 2 (if it had any)." As opposed to this approach where you throw functions in a set and have to check each function for what happens when

Reply:
done